### PR TITLE
Deprecate cents in favour of subunits

### PR DESCRIPTION
**Pull request by elfassy**
# Why
cents are not currency aware, which is especially important for currencies that have no minor units like the japanese yen.

re https://github.com/Shopify/shopify/issues/121143

# What
- deprecate `money.cents`
- modify allocation so it allocates subunits correctly
- added tests to verify specifically JPY

### CODE REVIEW

File: lib/money/money.rb [hunk 1]
```diff
@@ -70,7 +70,7 @@ def initialize(value = 0, currency = nil)
     raise ArgumentError if value.respond_to?(:nan?) && value.nan?
     @currency = Helpers.value_to_currency(currency)
     @value = Helpers.value_to_decimal(value).round(2)
```
Comment (sunblaze) [inline lib/money/money.rb]: Don't we also need to not always round the decimal value to 2 as well?

Comment (elfassy) [inline lib/money/money.rb]: was going to do that with #73 but yea this won't work with 3 digit currencies. Will cherry-pick that commit here

File: spec/money_spec.rb [hunk 2]
```diff
@@ -341,6 +343,13 @@
     end
   end
 
+  describe "value" do
+    it "rounds to the number of minor units provided by the currency" do
+      expect(Money.new(1.1111, 'USD').value).to eq(1.11)
```
Comment (bdewater) [inline spec/money_spec.rb]: Add a three-decimal currency for completeness?

**Comment by elfassy** [comment]
@christianblais @SebastianSzturo This PR will now correctly round money objects. This might have an impact on `JPY` (for example: https://github.com/Shopify/shopify/issues/124792).

**Pull request merged**